crates/biome_js_analyze/src/lint/nursery/no_secrets.rs

```diff
@@ -155,119 +223,284 @@ struct SensitivePattern {
+    let mut lowercase_count = 0;
+    let mut digit_count = 0;
+    let mut symbol_count = 0;
+    let mut case_switches = 0;
+    let mut previous_char_was_upper = false;
+
     for count in freq.iter() {
         if *count > 0 {
             let p = *count as f64 / len as f64;
-            entropy -= p * p.log2();
+            shannon_entropy -= p * p.log2();
+        }
+    }
+
+    // Letter classification and case switching
+    for (i, c) in data.chars().enumerate() {
+        if c.is_ascii_alphabetic() {
+            letter_count += 1;
+            if c.is_uppercase() {
+                uppercase_count += 1;
+                if i > 0 && !previous_char_was_upper {
+                    case_switches += 1;
+                }
+                previous_char_was_upper = true;
+            } else {
+                lowercase_count += 1;
+                if i > 0 && previous_char_was_upper {
+                    case_switches += 1;
+                }
+                previous_char_was_upper = false;
+            }
+        } else if c.is_ascii_digit() {
+            digit_count += 1;
+        } else if !c.is_whitespace() {
+            symbol_count += 1;
         }
     }
 
-    entropy
+    // Adjust entropy: case switches and symbol boosts
+    let case_entropy_boost = if uppercase_count > 0 && lowercase_count > 0 {
+        (case_switches as f64 / letter_count as f64) * 2.0
+    } else {
+        0.0
+    };
+
+    let symbol_entropy_boost = if symbol_count > 0 {
+        symbol_count as f64 / len as f64
+    } else {
+        0.0
+    };
+
+    let digit_entropy_boost = if digit_count > 0 {
+        digit_count as f64 / len as f64
+    } else {
+        0.0
+    };
+
+    let adjusted_entropy = shannon_entropy
+        + (case_entropy_boost * 2.5)
+        + (symbol_entropy_boost * 1.5)
+        + digit_entropy_boost;
+
+    // Apply exponential scaling to avoid excessive boosting for long, structured tokens
+    apply_exponential_entropy_scaling(adjusted_entropy, len, base_threshold, scaling_factor)
+}
+
+/*
+    A simple mechanism to scale entropy as the string length increases, the reason being that
+    large length strings are likely to be secrets.
+    TODO: However, at some point there should definitely be a cutoff i.e. 100 characters, because it's
+    probably base64 data or something similar at that point.
+    This was taken from GPT, and I sadly couldn't find references for it.
+*/
+fn apply_exponential_entropy_scaling(
+    entropy: f64,
+    token_length: usize,
+    base_threshold: f64,
+    scaling_factor: f64,
+) -> f64 {
+    // We will apply a logarithmic dampening to prevent excessive scaling for long tokens
+    let scaling_adjustment = (token_length as f64 / scaling_factor).ln();
+    base_threshold + entropy * scaling_adjustment
+}
+
+fn split_into_tokens(value: &str) -> Vec<&str> {
+    let delimiters = [' ', '\t', '\n', '.', ',', ';', ':', '/', '-', '_', '@'];
+    value
+        .split(|c| delimiters.contains(&c))
```

Comment:
You can directly pass the list of separators.
```suggestion
    value
        .split([' ', '\t', '\n', '.', ',', ';', ':', '/', '-', '_', '@'])
```